TestProjects/Navigation/RoutingExample.Core/ViewModels/TestCViewModel.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using MvvmCross.Core.Navigation;
+using MvvmCross.Core.ViewModels;
+using MvvmCross.Platform;
+using RoutingExample.Core.ViewModels;
+
+namespace RoutingExample.Core.ViewModels
+{
+    public class TestCViewModel
+        : MvxViewModel<User, User>
+    {
+        private readonly IMvxNavigationService _navigationService;
+        public TestCViewModel(IMvxNavigationService navigationService)
+        {
+            _navigationService = navigationService;
+        }
+
+        private string _parameter;
+        public string Parameter
+        {
+            get { return _parameter; }
+            set { SetProperty(ref _parameter, value); }
+        }
+
+        public void Init(string viewmodelcparameter)
```

Comment:
I will take a look into creating an interface that can have typed parameters too and create another PR